drivers/scheduler/k8s/k8s.go

```diff
@@ -200,7 +206,28 @@ func (k *K8s) ParseSpecs(specDir string) ([]interface{}, error) {
 		}
 		defer file.Close()
 
-		reader := bufio.NewReader(file)
+		specTemplate := template.New("spectemplate")
+		out, err := ioutil.ReadAll(file)
+		if err != nil {
+			return nil, err
+		}
+
+		var buff bytes.Buffer
+		var ioReader io.Reader = &buff
+
+		// in case we're not able to parse the file, warn the user and fallback to file reading
+		if _, err = specTemplate.Parse(string(out)); err != nil {
+			logrus.Warnf("failed to parse template. Cause: %v", err)
+			ioReader = file
+		}
+
+		// in case we're not able to execute the file, warn the user and fallback to file reading
+		if err = specTemplate.Execute(&buff, k.ExternalOpts); err != nil {
+			logrus.Warnf("failed to execute template. Cause: %v", err)
+			ioReader = file
+		}
+
+		reader := bufio.NewReader(ioReader)
 		specReader := yaml.NewYAMLReader(reader)
```

Comment:
I would prefer to have a file with all variables related to specs and then parse them and substitute to appropriate values rather than add them to structs.